lib/puppet/provider/service/systemd.rb

```diff
@@ -9,14 +11,8 @@
 
   commands :systemctl => "systemctl"
 
-  if Facter.value(:osfamily).downcase == 'debian'
-    # With multiple init systems on Debian, it is possible to have
-    # pieces of systemd around (e.g. systemctl) but not really be
-    # using systemd.  We do not do this on other platforms as it can
-    # cause issues when running in a chroot without /run mounted
-    # (PUP-5577)
-    confine :exists => "/run/systemd/system"
-  end
+  next unless Puppet::FileSystem.exist?('/proc/1/cmdline')
+  confine :true => Puppet::FileSystem.read('/proc/1/cmdline').include?('systemd')
```

Comment:
On my Ubuntu 18.04 system, `/proc/1/cmdline` is `/sbin/initsplash`, and doesn't contain the string `systemd`.

It does look like the `/proc/1/exe` symlink does point to `/lib/systemd/systemd`, though. Perhaps that's the better thing to try to confine on?

Reply:
Not every distro has `/lib/systemd/systemd` as pointed out above.
However in my very unscientific tests the distro subset mentioned over a year ago all seem to have `/proc/1/exe` available, making it a valid candidate. Thanks for that!

Patch coming right up.